keps/sig-scheduling/4815-dra-partitionable-devices/README.md

```diff
@@ -252,222 +252,465 @@ non-goals of this KEP.
+  // order. Conflicting properties from multiple mixins are taken from the
+  // last mixin listed that contains them. Properties set on the capacity pool will
+  // always override properties from mixins.
+  //
+  // The mixins referenced here must be defined in the same
+  // ResourceSlice.
+  //
+  // The maximum number of mixins that can be included is 8.
+  //
+  // +optional
+  // +listType=atomic
+  Includes []CapacityPoolMixinRef
+
+  // Capacity defines the set of capacities for this capacity pool
+  // The name of each capacity must be unique in that set.
+  //
+  // To ensure this uniqueness, capacities defined by the vendor
+  // must be listed without the driver name as domain prefix in
+  // their name. All others must be listed with their domain prefix.
+  //
+  // Capacities listed here will always take precedence over any included
+  // from a mixin.
+  //
+  // The maximum number of capacities is 32.
+  //
+  // +required
+  Capacity map[QualifiedName]DeviceCapacity
 }
 
 // DeviceMixinRef defines a reference to a device mixin.
 type DeviceMixinRef struct {
-	// Name refers to the name of a device mixin in the pool.
-	//
-	// +required
-	Name string `json:"name"`
+  // Name refers to the name of a device mixin in the pool.
+  //
+  // +required
+  Name string
+}
+
+// CapacityPoolMixinRef defines a reference from a capacity pool
+// to a capacity pool mixin.
+type CapacityPoolMixinRef struct {
+  // Name is the name of a CapacityPoolMixin.
+  //
+  // +required
+  Name string
 }
 
-// DeviceRef defines a reference to a device.
-type DeviceRef struct {
-	// Name refers to the name of a device in the pool.
-	//
-	// +required
-	Name string `json:"name"`
+type DeviceCapacityConsumptionMixinRef struct {
+  // Name is the name of a DeviceCapacityConsumptionMixin.
+  //
+  // +required
+  Name string
+}
+
+// DeviceCapacityConsumption defines a set of capacities that
+// a device will consume from a capacity pool.
+type DeviceCapacityConsumption struct {
+  // CapacityPool defines the capacity pool from which the
+  // capacities defined (either directly or through a mixin)
+  // will be consumed from.
+  //
+  // +required
+  CapacityPool string
+  
+  // Includes defines a list of references to DeviceCapacityConsumptionMixins.
+  // The capacities listed in these will be included in among the
+  // capacities that will be consumed by the device.
+  //
+  // Capacities listed directly will override any capacities coming
+  // from mixins.
+  //
+  // The maximum number of mixins that can be included is 8.
+  //
+  // +optional
+  Includes []DeviceCapacityConsumptionMixinRef
+
+  // Capacity defines the capacity that will be consumed by
+  // the device.
+  //
+  // Capacities listed here will override any capacities that
+  // are also defined in any of the referenced mixins.
+  //
+  // The maximum number of capacities is 32.
+  //
+  // +optional
+  Capacity DeviceCapacity
```

Comment:
Should we also enforce that the "effective" number of capacities for a specific device (i.e. the number of capacities defined directly plus any included through mixins) also remain below the max number? I think we probably should, but just wanted to see if anyone has an opinion on this.